src/main/groovy/org/shipkit/gradle/exec/ExecCommand.java

```diff
@@ -0,0 +1,81 @@
+package org.shipkit.gradle.exec;
+
+import org.gradle.api.Action;
+import org.gradle.process.ExecResult;
+import org.gradle.process.ExecSpec;
+
+import java.util.Collection;
+
+/**
+ * Object that contains information about the executable command line.
+ * It has: the command line arguments, action that configures the execution,
+ * action that is triggered when the execution is complete.
+ */
+public class ExecCommand {
+
+    private final static Action NO_OP_ACTION = new Action() {
+        public void execute(Object o) {}
+    };
+
+    private final static Action ENSURE_SUCCEEDED_ACTION = new Action<ExecResult>() {
+        public void execute(ExecResult result) {
+            //TODO offer cleaner exception than the one by default offered by Gradle
+            result.assertNormalExitValue();
+        }
+    };
+
+    private final Collection<String> commandLine;
+    private final Action<ExecSpec> setupAction;
+    private final Action<ExecResult> resultAction;
+
+    /**
+     * Generic command line to be executed
+     *
+     * @param commandLine command line to be executed
+     * @param setupAction action that configures the command line execution
+     * @param resultAction action that is triggered after the command line was executed
+     */
+    public ExecCommand(Collection<String> commandLine, Action<ExecSpec> setupAction, Action<ExecResult> resultAction) {
+        this.commandLine = commandLine;
```

Comment:
Maybe a defensive copy of the collection could be made (to make it more immutable)?